tests/devices/test_default_mixed.py

```diff
@@ -235,6 +235,15 @@ def test_none_state(self, nr_wires):
 
         assert dev.analytic_probability() is None
 
+    def test_probability_not_negative(self, nr_wires):
+        """Test that probabilities are always real"""
+        dev = qml.device("default.mixed", wires=nr_wires)
+        dev._state = np.zeros([2 ** nr_wires, 2 ** nr_wires])
+        dev._state[0, 0] = 1
```

Comment:
Should we also do a check to ensure the probabilities are bounded by 1? I imagine if errors can cause 0 to be ~ -5e-17, then it could also take 1 to 1 + 5e-17?

Reply:
I think that for any quantum simulator, probabilities will never add to exactly 1 due to numerical precision. But this is not a real problem since it just means that any quantity we compute will suffer from numerical imprecision.

However, negative probabilities are a real issue since they may throw errors when used later. So I think the change in this PR is a good one, but no need to check if probabilities are bounded by one

Reply:
Ahh sounds good !